dig.go

```diff
@@ -664,9 +685,12 @@ func (n *node) Call(c containerStore) error {
 	}
 
 	receiver := newStagingContainerWriter()
-	results := reflect.ValueOf(n.ctor).Call(args)
-	if err := n.resultList.ExtractList(receiver, results); err != nil {
-		return errConstructorFailed{Func: n.location, Reason: err}
+	// Do not call providers if we are in dry mode and  dont want anything to run.
+	if !c.Dry() {
```

Comment:
Oh, nice, this is exactly what I had messaged you offline about — that provides would still get executed — but this addition already addresses that.